_posts/2016-01-29-back-to-business.md

```diff
@@ -0,0 +1,58 @@
+
+This is my 12th working day after quite a long break, a little over 1,5 years maternity leave. Before that I had been working in KIRRE project since 2010. In KIRRE I worked as a software developer and had my hands on both in back end and front end. KIRRE project was dealing with registry of legal confirmations to real properties. 
+
+##Jump to the Moving Train
+
+After working a long time in the same project, it seemed to be reasonable to change the project, from real properties to a completely different world, railways. LIIKE project has been started in 2008 in co-operation with Liikennevirasto. LIIKE projects deals with managing the capacity of railways and supervising and controlling the actual traffic. The project contains maintenance and support, but also further development.
+
+It's always a challenge to familiarize oneself with a new business branch, ecspecially when the project has already been active for a long time and the business logic is full of little but still essential details. A huge amount of information is just waiting to be digested.
+
+LIIKE project consists of several applications developed using various technologies. The back end is implemented using Java. In the front end there are both web and desktop applications. 
+
+##A Story of One Jira Issue
+
+Our backlog of tasks to be implemented is stored in Jira. One can pick a task from the backlog and start implementing the task. As I'm a novice in the project the project manager has picked suitable tasks for me. Not too fierce in the beginning. Today I started with a task about showing an icon and tooltip in the user interface when there is something worth of noticing when a train assembly is inspected, e.g. the train is longer than it should.
+
+We use IntelliJ IDEA as development environment. I've cloned the project codes from remote repository to my local repository using Git. IntelliJ IDEA feels quite familiar after using Eclipse but I'm just hoping that they had similar keyboard shortcuts. I still try to open types with Ctrl + Shift + T and so on. Git is also a new acquintance to me. I still feel a little bit clumsy with all the branches but I like the idea to be able to make commits just in my own repository.
+
+The task in hand needs to be done in Reaali application which has user interface implemented using Java Swing. I have not used Swing since 2005 but there is some correspondence to Apache Wicket which was used in the previous project. However, when using Swing, html-tags need to be used only when formatting tool tip texts. 
```

Comment:
has _a_ user interface

correspondence = kanssakäyminen (ensisijainen merkitys) ... vaihtaisin esim.  "but it has some similarity to Apache Wicket"